### ISSUE TITLE
[Request] Add proxy support or at least custom DNS to revive online gaming

### ISSUE THREAD

**Issue by MJaoune** [libretro/RetroArch#5508]
## Description
Please add proxy support or custom DNS for connections done inside cores, so that the DNS (Fan driven) would redirect the connection to a fan-driven game server rather than the game's official servers which would most likely be down. Not all devices that run retroarch support custom DNS, or maybe it would require intermediate steps.

### Expected behavior
To be able to choose a custom DNS from RetroArch settings so that games run on different cores would establish a connection using it.

**Comment by fr500** [comment]
RA having custom DNS wouldn't help at all.
The cores are programs and if they have networking they'll do it directly
via syscalls.

**Comment by orbea** [comment]
So this would need to be implemented inside the cores?

**Comment by fr500** [comment]
Yeah. Best case scenario there could be an env callback so the frontend can provide it.
